escrow/transactions/user/add_entry.cdc

```diff
@@ -0,0 +1,35 @@
+import NonFungibleToken from "../../../contracts/NonFungibleToken.cdc"
+import AllDay from "../../../contracts/AllDay.cdc"
+import Escrow from "../../../contracts/AllDay.cdc"
+
+transaction(leaderboardName: String, nftID: UInt64) {
+    let nft: @NonFungibleToken.NFT
+    let escrowRef: &Escrow.Collection
+    let receiver: Capability<&{NonFungibleToken.CollectionPublic}>
+
+    prepare(signer: AuthAccount, admin: AuthAccount) {
+        // Borrow a reference to the user's NFT collection as a Provider
+        let collectionRef = signer.borrow<&{NonFungibleToken.Provider}>(
+            from: AllDay.CollectionStoragePath
+        ) ?? panic("Could not borrow NFT collection reference")
+
+        // Borrow a reference to the user's NFT collection as a Receiver.
+        self.receiver = signer.getCapability<&{NonFungibleToken.CollectionPublic}>(AllDay.CollectionPublicPath)!
+
+        // Withdraw the NFT from the user's collection
+        self.nft <- collectionRef.withdraw(withdrawID: nftID)
+
+        // Borrow a reference to the Escrow Collection resource from the collection account
+        self.escrowRef = admin.borrow<&Escrow.Collection>(from: Escrow.CollectionStoragePath)
```

Comment:
Not sure this was intentional, but do we actually want to required a multi-sig tx here?
This will likely make it a bit more complex to actually run this tx.
Would it be safe to safe to have addEntry be part of the ICollectionPublic interface, that way we don't need an admin signature to run this tx?

Reply:
Great callout! I was trying to have this implemented beforehand and didn't see a way. I do now. Updated!